### PR TITLE
Test(pytest) start aut

### PR DESCRIPTION
**Pull request by StateOf-Vlado**
Fixes ([Prepare structure for "squish driver"](https://github.com/status-im/desktop-qa-automation/issues/67))
Fixes ([Implement managing squishserver during a test](https://github.com/status-im/desktop-qa-automation/issues/68))
Fixes ([Implement setup, start, and stop the application with custom user data](https://github.com/status-im/desktop-qa-automation/issues/69))
Fixes ([Prepare the Main Window fixture](https://github.com/status-im/desktop-qa-automation/issues/70))
Fixes ([Save screenshot on failed test](https://github.com/status-im/desktop-qa-automation/issues/71))

### Manage squish server and AUT

- Wrappers for UI elements moved from the previous version framework
- Methods for starting/stopping the squish server added
- Methods for launching, stopping, and detaching the application added

I found, for some versions of PyCharm, an environment variable, which we add (SQUISH_DIR, PYTHONPATH, LD_LIBRARY_PATH) replaced by PyCharm, so more stable way to add environment variable, to set it specifically for the test:
image

**Before testing, please update the requirements: step 6 in the README file**

Test (execute for CI app build)
1)Set a breakpoint after driver.context.detach() in test_start_aut and execute the test in debug mode.
image

2)When the execution stops on the breakpoint, open squish IDE, and open suite settings
3) add Attachable AUT in 'Manage AUT' with the parameters:
- Name: Status
- Host: localhost
- Port: 61500
image

4) Select 'No Application' in AUT settings
5) Save changes and close suite settings.
6) Start AUT by button 
image
on the toolbar:
image

7) In the 'Launch Settings' window select attachable Status and click  <OK> button
image

Elements should be visible for the Pick tool in Squish IDE

We have now two methods to start the test application:
- by subprocesses (we will use it only for local run, to explore application inside)
- by squish startApplication (attach/detach mode not available)
Start AUT method can be selected by ATTACH_MODE key in ._local.py file

### CODE REVIEW

**Review by noeliaSD — COMMENTED** [review]
Some comments added! 

Next step is following the steps you wrote and test it. I'll let you know the result!

File: test/ui-pytest/README.md
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/README.md:1]: Fixed

File: test/ui-pytest/conftest.py
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/conftest.py:1]: It is a default pytest file, so we have to use it

File: test/ui-pytest/configs/testpath.py
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/configs/testpath.py:1]: if we rename it, pytest will look for tests in this file.
pytest parse all files that start with test in the name

File: test/ui-pytest/driver/objects_access.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import logging
+
+import object
+
+_logger = logging.getLogger(__name__)
+
+
+def walk_children(parent, depth: int = 1000):
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/driver/objects_access.py:8]: renamed

File: test/ui-pytest/gui/elements/base_object.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    def __str__(self):
+        return f'{type(self).__qualname__}({self.symbolic_name})'
+
+    @property
+    def object(self):
+        return driver.waitForObject(self.real_name, configs.timeouts.UI_LOAD_TIMEOUT_MSEC)
+
+    @property
+    def entity(self):
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/gui/elements/base_object.py]: deleted

File: test/ui-pytest/gui/elements/base_object.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    @property
+    def height(self) -> int:
+        return int(self.bounds.height)
+
+    @property
+    def center(self):
+        return self.bounds.center()
+
+    @property
+    def is_enabled(self) -> bool:
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/gui/elements/base_object.py:45]: Tru, good idea, fixed

File: test/ui-pytest/gui/elements/base_object.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    @property
+    def is_selected(self) -> bool:
+        return self.object.selected
+
+    @property
+    def is_checked(self) -> bool:
+        return self.object.checked
+
+    @property
+    def is_visible(self) -> bool:
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/gui/elements/base_object.py:57]: fixed

File: test/ui-pytest/tests/fixtures/aut.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+@pytest.fixture
+def user_data(request) -> system_path.SystemPath:
+    user_data = configs.testpath.STATUS_DATA / f'app_{datetime.now():%H%M%S_%f}' / 'data'
+    if hasattr(request, 'param'):
+        system_path.SystemPath(request.param).copy_to(user_data)
+    yield user_data
+
+
+@pytest.fixture
+def user_data_two(request, user_data) -> system_path.SystemPath:
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply test/ui-pytest/tests/fixtures/aut.py:33]: We don't need it now, so I removed it

**Review by noeliaSD — APPROVED** [review]
As agreed, still not working on my local machine so cannot test it propertly but approved to not block the `pytest works`. This pr is not affecting the existing e2e tests.

**Review by anastasiyaig — APPROVED** [review]
